mw_virtual_endpoint.go

```diff
@@ -223,13 +232,10 @@ func (d *VirtualEndpoint) ServeHTTPForCache(w http.ResponseWriter, r *http.Reque
 	newResponse.Body = ioutil.NopCloser(&bodyBuffer)
 	copiedRes.Body = ioutil.NopCloser(bodyBuffer2)
 
-	d.HandleResponse(w, newResponse, session)
+	HandleForcedResponse(w, newResponse, session)
 
 	// Record analytics
-	go d.sh.RecordHit(r, 0, newResponse.StatusCode, copiedRequest, copiedResponse)
-
-	return copiedRes
-
+	return copiedRes, newResponse.StatusCode
```

Comment:
Do you need the two return values? `StatusCode` is also part of the response copy.